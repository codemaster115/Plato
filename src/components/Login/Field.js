import React from 'react';
import { FormControl, Input, InputLabel, Link } from '@material-ui/core';
import styled from 'styled-components';

const StyledLink = styled(Link)`
  position: relative;
  left: 50%;
  transform: translateX(-50%);
  margin: 5px 0px 10px !important;
`;

export default class ButtonComponent extends React.Component {
  onEmailChange = e => {
    this.props.userNameChange(e.target.value);
  };

  onPasswordChange = e => {
    this.props.passwordChange(e.target.value);
  };

  render() {
    return (
      <div>
        <FormControl margin="normal" required fullWidth>
          <InputLabel htmlFor="email">Email Address</InputLabel>
          <Input
            id="email"
            name="email"
            autoComplete="email"
            autoFocus
            onChange={this.onEmailChange}
          />
        </FormControl>
        <FormControl margin="normal" required fullWidth>
          <InputLabel htmlFor="password">Password</InputLabel>
          <Input
            name="password"
            type="password"
            id="password"
            autoComplete="current-password"
            onChange={this.onPasswordChange}
          />
        </FormControl>
        <StyledLink
          component="button"
          variant="body2"
          onClick={() => {
            this.onValChange();
          }}
        >
          Forgot your password?
        </StyledLink>
      </div>
    );
  }
}
