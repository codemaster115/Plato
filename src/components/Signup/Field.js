import React from 'react';
import {
  FormControl, 
  Input, 
  InputLabel
 } from '@material-ui/core';

class ButtonComponent extends React.Component {
  render() {
    return (
      <div>
         <FormControl margin="normal" required fullWidth>
          <InputLabel htmlFor="name">First Name</InputLabel>
          <Input id="first_name" name="first_name" autoFocus />
        </FormControl>
        <FormControl margin="normal" required fullWidth>
          <InputLabel htmlFor="name">Last Name</InputLabel>
          <Input id="last_name" name="last_name" />
        </FormControl>
        <FormControl margin="normal" required fullWidth>
          <InputLabel htmlFor="email">Email</InputLabel>
          <Input id="email" name="email" autoComplete="email" />
        </FormControl>
        <FormControl margin="normal" required fullWidth>
          <InputLabel htmlFor="password">Password</InputLabel>
          <Input name="password" type="password" id="password" autoComplete="current-password" />
        </FormControl>
        <FormControl margin="normal" required fullWidth>
          <InputLabel htmlFor="confirm_password">Confirm your Password</InputLabel>
          <Input name="confirm_password" type="password" id="confirm_password" autoComplete="current-password" />
        </FormControl>
      </div>
    );
  }
}

export default ButtonComponent;
