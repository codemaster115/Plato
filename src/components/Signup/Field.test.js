import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Input } from '@material-ui/core';

import Field from './Field';

describe('Signup Field', () => {
  it('Compare snapshots', () => {
    const Component = renderer.create(<Field />);

    const Tree = Component.toJSON();
    expect(Tree).toMatchSnapshot();
  });

  it('Renders without crashing', () => {
    mount(<Field />);
  });

  it('Renders 5 input field', () => {
    const wrapper = shallow(<Field />);
    expect(wrapper.find(Input)).toHaveLength(5);
  });
});
