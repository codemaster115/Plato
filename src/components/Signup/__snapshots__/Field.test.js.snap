// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Field Compare snapshots 1`] = `
<div>
  <div
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
  >
    <label
      className="MuiFormLabel-root-16 MuiFormLabel-required-21 MuiInputLabel-required-9 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
      data-shrink={false}
      htmlFor="name"
    >
      First Name
      <span
        className="MuiFormLabel-asterisk-22"
      >
         *
      </span>
    </label>
    <div
      className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-formControl-37 MuiInput-formControl-24"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={true}
        className="MuiInputBase-input-46 MuiInput-input-31"
        disabled={false}
        id="first_name"
        name="first_name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
  >
    <label
      className="MuiFormLabel-root-16 MuiFormLabel-required-21 MuiInputLabel-required-9 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
      data-shrink={false}
      htmlFor="name"
    >
      Last Name
      <span
        className="MuiFormLabel-asterisk-22"
      >
         *
      </span>
    </label>
    <div
      className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-formControl-37 MuiInput-formControl-24"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input-46 MuiInput-input-31"
        disabled={false}
        id="last_name"
        name="last_name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
  >
    <label
      className="MuiFormLabel-root-16 MuiFormLabel-required-21 MuiInputLabel-required-9 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
      data-shrink={false}
      htmlFor="email"
    >
      Email
      <span
        className="MuiFormLabel-asterisk-22"
      >
         *
      </span>
    </label>
    <div
      className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-formControl-37 MuiInput-formControl-24"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="email"
        className="MuiInputBase-input-46 MuiInput-input-31"
        disabled={false}
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
  >
    <label
      className="MuiFormLabel-root-16 MuiFormLabel-required-21 MuiInputLabel-required-9 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
      data-shrink={false}
      htmlFor="password"
    >
      Password
      <span
        className="MuiFormLabel-asterisk-22"
      >
         *
      </span>
    </label>
    <div
      className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-formControl-37 MuiInput-formControl-24"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="current-password"
        className="MuiInputBase-input-46 MuiInput-input-31 MuiInputBase-inputType-49 MuiInput-inputType-34"
        disabled={false}
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="password"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
  >
    <label
      className="MuiFormLabel-root-16 MuiFormLabel-required-21 MuiInputLabel-required-9 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
      data-shrink={false}
      htmlFor="confirm_password"
    >
      Confirm your Password
      <span
        className="MuiFormLabel-asterisk-22"
      >
         *
      </span>
    </label>
    <div
      className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-formControl-37 MuiInput-formControl-24"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="current-password"
        className="MuiInputBase-input-46 MuiInput-input-31 MuiInputBase-inputType-49 MuiInput-inputType-34"
        disabled={false}
        id="confirm_password"
        name="confirm_password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="password"
      />
    </div>
  </div>
</div>
`;
